File: rockpaper/rockpaper.cpp
#include <bits/stdc++.h>
#define rand(l, h) (rand() % (h - l + 1) + l)
using namespace std;
unordered_map<string, int> m = {{"Rock", 0}, {"Paper", 1}, {"Scissors", 2}};
int beat[3] = {2, 0, 1};
int main()
{
    while (1)
    {
        cout << "Rock, Paper, or Scissors?" << endl;
        srand(time(0));
        int cc = rand(0, 2);
        string in;
        cin >> in;
        if (m.find(in) == m.end())
        {
            cout << "Invalid input" << endl;
            continue;
        }
        int uc = m[in];
        if (uc == cc)
        {
            cout << "Tie" << endl;
        }
        else if (beat[uc] == cc)
        {
            cout << "You win" << endl;
        }
        else if (beat[cc] == uc)
        {
            cout << "You lost" << endl;
        }
    }
    return 0;
}

File: rockpaper/rockpaper.py
from random import randint

while True:
    computer_choice = number = randint(0, 2)  # 0=Rock 1=Paper 2=Scissors
    choice = input("Rock, Paper, or Scissors?")
    if choice == "Rock":
        if computer_choice == 0:
            print("Tie")
        elif computer_choice == 1:
            print("You lose. Paper -> Rock")
        elif computer_choice == 2:
            print("You win. Rock -> Scissors")
    elif choice == "Paper":
        if computer_choice == 0:
            print("You win. Paper -> Rock")
        elif computer_choice == 1:
            print("Tie")
        elif computer_choice == 2:
            print("You lose. Scissors -> Paper")
    elif choice == "Scissors":
        if computer_choice == 0:
            print("You lose. Rock -> Scissors")
        elif computer_choice == 1:
            print("You win. Scissors -> Paper")
        elif computer_choice == 2:
            print("Tie")
    else:
        print("Invalid input")
